### PR TITLE
[CDR-1451] Basic ItemTag support

### PR DESCRIPTION
**Pull request by alexlehn**
# Changes

Support for ITEM_TAG crud operation on an experimental REST endpoint

# Pre-Merge checklist

- [ ] New code is tested
- [ ] Present and new tests pass
- [ ] Documentation is updated
- [ ] The build is working without errors
- [ ] No new Sonar issues introduced
- [ ] Changelog is updated
- [ ] Code has been reviewed

### CODE REVIEW

File: rest-openehr/src/main/java/org/ehrbase/rest/openehr/experimental/ItemTagController.java [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+            @PathVariable(value = "ehr_id") String ehrIdString,
+            @PathVariable(value = "versioned_object_uid") String versionedObjectUid,
+            @RequestBody Collection<ItemTagDto> itemTags) {
+
+        return upsertItemTags(
+                accept,
+                prefer,
+                ehrIdString,
+                versionedObjectUid,
+                ItemTag.ItemTagRMType.EHR_STATUS,
```
Comment (vidi42) [inline rest-openehr/src/main/java/org/ehrbase/rest/openehr/experimental/ItemTagController.java:90]: I believe a 400 Bad Request should be thrown if someone would want to create a COMPOSITION tag trough this endpoint, by setting the `target_type` to `COMPOSITION`. (Also applies to the composition endpoint)

File: rest-openehr/src/main/java/org/ehrbase/rest/openehr/experimental/ItemTagController.java [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+            @RequestHeader(value = PREFER, required = false) String prefer,
+            @PathVariable(value = "ehr_id") String ehrIdString,
+            @PathVariable(value = "versioned_object_uid") String versionedObjectUid,
+            @RequestBody Collection<ItemTagDto> itemTags) {
+
+        return upsertItemTags(
+                accept,
+                prefer,
+                ehrIdString,
+                versionedObjectUid,
```
Comment (vidi42) [inline rest-openehr/src/main/java/org/ehrbase/rest/openehr/experimental/ItemTagController.java:89]: Setting a different `target` than the `versionedObjectUid` through the DTO should maybe result in a 400 Bad Request. Same for the ownerId (Also applies to the composition endpoint)

**Review by vidi42 — CHANGES_REQUESTED** [review]

**Comment by alexlehn** [comment]
@vidi42 I've Added a sanity check for the `target_type` to the Tag service https://github.com/ehrbase/ehrbase/pull/1343/commits/9887b0fd19059abe5b84c4771abd28e83d378964

File: api/src/main/java/org/ehrbase/api/dto/experimental/ItemTagDto.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonRootName;
+import java.util.UUID;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import org.ehrbase.api.service.experimental.ItemTag;
+
+/**
+ * ItemTag DTO use as a PUT or DELETE input as well as output for GET.
```
Comment (vmueller-vg) [inline api/src/main/java/org/ehrbase/api/dto/experimental/ItemTagDto.java:28]: "use" -> "used"

File: configuration/src/main/java/org/ehrbase/configuration/config/SwaggerConfiguration.java [hunk 4]
```diff
@@ -68,6 +69,15 @@ public GroupedOpenApi actuatorApi() {
                 .build();
     }
 
+    @Bean
+    @ConditionalOnProperty(name = "ehrbase.rest.experimental.tags.enabled", havingValue = "true")
+    public GroupedOpenApi experimentalApi() {
+        return GroupedOpenApi.builder()
+                .group("6. Experimental API")
+                .pathsToMatch("/rest/experimental/**")
```
[the remark this answers is not in the document]
Comment (alexlehn) [reply configuration/src/main/java/org/ehrbase/configuration/config/SwaggerConfiguration.java]: thx ... forgot about this, it's now [fixed](https://github.com/ehrbase/ehrbase/pull/1343/commits/461d56e982a960d568ab647f75bc98b79f2a94a7#diff-53f034c499b009148c7f5b6c620e35be4afb8a7829ebdbb01a9001d22f6a4048R75-R85)

File: api/src/main/java/org/ehrbase/api/dto/experimental/ItemTagDto.java (hunk 3 above)
Comment (alexlehn) [reply api/src/main/java/org/ehrbase/api/dto/experimental/ItemTagDto.java]: fixed https://github.com/ehrbase/ehrbase/pull/1343/commits/3244f6842e01261da775aa204924a0bf1945c171

File: rest-openehr/src/main/java/org/ehrbase/rest/openehr/experimental/ItemTagController.java (hunk 1 above)
Comment (vmueller-vg) [reply rest-openehr/src/main/java/org/ehrbase/rest/openehr/experimental/ItemTagController.java]: this is checked in the service

**Pull request merged**